### PR TITLE
feat: nft table filter by network

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR adds support for filtering NFTs by network on the NFT table, programmatic on the chains of the available NFTs.

This is in addition to the search match sorting.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Low to none, though, ensure search match sorting is still working as expected

## Testing

- Search filtering is still working
- Filter by network is working
- Combining search filtering and network filtering properly narrows the intersection of the two

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4563** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4563" target="_blank">Graphite</a> 
    * **PR #4574** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4574" target="_blank">Graphite</a> 
      * **PR #4582** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4582" target="_blank">Graphite</a> 
        * **PR #4575** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4575" target="_blank">Graphite</a> 
          * **PR #4596** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4596" target="_blank">Graphite</a> 
            * **PR #4597** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4597" target="_blank">Graphite</a> 
              * **PR #4610** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4610" target="_blank">Graphite</a>  👈
              * **PR #4609** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4609" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4610?utm_source=stack-comment).

File: src/components/Nfts/NftNetworkFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  availableChainIds: ChainId[]
+}
+
+export const NftNetworkFilter = ({
+  setFilters,
+  resetFilters,
+  hasAppliedFilter = false,
+  availableChainIds,
+}: NftNetworkFilterProps) => {
+  const [isOpen, setIsOpen] = useState(false)
```
Comment (gomesalexandre) [inline src/components/Nfts/NftNetworkFilter.tsx:43]: May or may not be able to use a `useToggle` implementation but let's be safe, this is pretty much a stripped down, NFT network selection repurposed implementation of `TransactionHistoryFilter` - the goal is to get this working and merged, not cleanup the `<TransactionHistoryFilter />` (and this one while at it) or abstract things away

File: src/components/Nfts/NftNetworkFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+              variant='ghost-filled'
+              leftIcon={<IoOptionsOutline size='1.5em' />}
+              onClick={() => setIsOpen(state => !state)}
+            >
+              <Text translation='transactionHistory.filter' />
+            </Button>
+            <IconButton
+              variant='ghost-filled'
+              colorScheme='blue'
+              aria-label={translate('transactionHistory.filters.resetFilters')}
```
Comment (gomesalexandre) [inline src/components/Nfts/NftNetworkFilter.tsx:99]: Could be namespaced to a nft namespaced translation but meh

File: src/components/Nfts/NftNetworkFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+              >
+                <Text translation='transactionHistory.filters.resetFilters' />
+              </Button>
+            </Flex>
+            <Divider />
+            <form onSubmit={handleSubmit(onSubmit)}>
+              <FilterGroup
+                name={FilterFormFields.Network}
+                control={control}
+                title='Network'
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/components/Nfts/NftNetworkFilter.tsx]: d164ae856ce1413c4ce0da167fcd52ac474c0318

**Review by kaladinlight — DISMISSED** [review]

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**